pkg/systemd/hw-detect.js

```diff
@@ -61,6 +63,58 @@ function findPCI(udevdb, info) {
     }
 }
 
+function findMemoryDevices(udevdb, info) {
+    const memoryArray = [];
+    const dmipath = '/devices/virtual/dmi/id';
+    if (!(dmipath in udevdb)) {
+        return;
+    }
+
+    const props = udevdb[dmipath];
+    // Systemd now exposes memory information in udev, introduced in systemd => 248
+    // https://github.com/systemd/systemd/blob/main/NEWS#L1713
+    if ('MEMORY_ARRAY_NUM_DEVICES' in props) {
+        for (let slot = 0; slot < props.MEMORY_ARRAY_NUM_DEVICES; slot++) {
+            let memorySize = props[`MEMORY_DEVICE_${slot}_SIZE`];
+            if (memorySize) {
+                memorySize = cockpit.format(_("$0 GB"), memorySize / 1024 / 1024 / 1024);
```

Comment:
unit confusion, this is GiB. Please use our standard [cockpit.format_bytes()](https://cockpit-project.org/guide/latest/cockpit-util.html) API for that, with an explicit `factor=1024` (as this is one of the very few cases where we actually *do* want base-2 units)